geospaas_harvesting/crawlers.py

```diff
@@ -377,3 +379,63 @@ def _get_resources_urls(self, xml):
             self._urls.append(entry['link'])
 
         return bool(entries)
+
+
+class FTPCrawler(WebDirectoryCrawler):
+    """
+    Crawler which returns the search results of an FTP, given the URL and search
+    terms
+    """
+    LOGGER = logging.getLogger(__name__ + '.FTPCrawler')
+
+    def __init__(self, root_url, username=None, password=None, fileformat=None,):
+        self.root_url = root_url
+        self.password = password
+        self.username = username
+        self.FILES_SUFFIXES = fileformat
+        self.set_initial_state()
+
+    def set_initial_state(self):
+        """
+        The `_urls` attribute contains URLs to the resources which will be returned by the crawler.
+        The `_to_process` attribute contains URLs to pages which need to be searched for resources.
+        """
+        self._urls = []
+        if not self.root_url.startswith('ftp://'):
+            raise ValueError("root url must start with 'ftp://' in the configuration file")
```

Comment:
@aperrin66 
it is inside the `set_initial_state` function which is called only from the constructor. It is ok, isn't it?

Reply:
`set_initial_state` is not only called in the constructor.
It's also used by the harvester to reinitialize the crawler so that the endless mode can work.

Reply:
Done in the next commit